tests/prague/eip7002_el_triggerable_withdrawals/test_modified_withdrawal_contract.py

```diff
@@ -0,0 +1,261 @@
+        ),
+        pytest.param(
+            [
+                *[
+                    single_withdrawal_with_custom_fee(i)
+                    for i in range(
+                        0,
+                        18,
+                    )
+                ],
+            ],
+            id="18_withdrawal_requests",
+        ),
+    ],
+)
+def test_extra_withdrawals_pseudo_contract(
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+    requests_list: List[WithdrawalRequest],
+):
+    """Test how clients were to behave when more than 16 withdrawals would be allowed per block."""
+    modified_code: Bytecode = Bytecode()
+    memory_offset: int = 0
+    amount_of_requests: int = 0
+
+    # Goal: Have contract return a bunch of withdrawal requests
+    #   Problem: EVM has no concept of withdrawal request, it just return bytes from memory
+    #       Problem: How to get __bytes__ from withdrawal request
+    #       Problem: How to know exact size of withdrawal requests byte representation
+    #       Problem: If size larger than 32 bytes how to split across multiple MSTOREs?
+
+    # what size does a withdrawal_request bytes representation have?
+    #   withdrawal_request.__bytes__:
+    #             bytes(self.source_address)            ->          20 bytes
+    #           + bytes(self.validator_pubkey)          ->          48 bytes
+    #           + self.amount.to_bytes(8, "little")     ->          8 bytes
+    #                                                   -> Total:   76 bytes
+    for withdrawal_request in requests_list:
+        withdrawal_request_chunk_1_3_32bytes: bytes = withdrawal_request.__bytes__()[:32]
+        withdrawal_request_chunk_2_3_32bytes: bytes = withdrawal_request.__bytes__()[32:64]
+        withdrawal_request_chunk_3_3_12bytes: bytes = withdrawal_request.__bytes__()[64:]
+
+        modified_code += Op.MSTORE(memory_offset, withdrawal_request_chunk_1_3_32bytes)
+        memory_offset += 32
+
+        modified_code += Op.MSTORE(memory_offset, withdrawal_request_chunk_2_3_32bytes)
+        memory_offset += 32
+
+        modified_code += Op.MSTORE(memory_offset, withdrawal_request_chunk_3_3_12bytes)
+        # memory_offset += 32  # MSTORE ALWAYS writes 32 bytes, no need to update memory offset tho
+
+        amount_of_requests += 1
+
+    modified_code += Op.RETURN(
+        0, 76 * amount_of_requests
+    )  # don't care about the zeroes added by MSTORE at [76,96] bytes
+
+    pre[Spec_EIP7002.WITHDRAWAL_REQUEST_PREDEPLOY_ADDRESS] = Account(
+        code=modified_code,
+        nonce=1,
+        balance=0,
+    )
+
+    # given a list of withdrawal requests construct a withdrawal request transaction
+    withdrawal_request_transaction = WithdrawalRequestTransaction(requests=requests_list)
+    # prepare withdrawal senders
+    withdrawal_request_transaction.update_pre(pre=pre)
+    # get transaction list
+    txs: List[Transaction] = withdrawal_request_transaction.transactions()
```

Comment:
Interesting thing is that, with this contract, we don't even have to add any transactions to the block, but I think it's ok for the sake of completeness.